### PR TITLE
feat: add testnode implementation

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

Closes https://github.com/celestiaorg/orchestrator-relayer/issues/55

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Comment by evan-forbes** [comment]
wow, this is really cool!

**Review by evan-forbes — DISMISSED** [review]
nice! took a second look, and this makes sense to me so far. LGTM, utACK :+1: :+1: 

 will defer to @rahulghangas for final approval

**Comment by SweeXordious** [comment]
CI will be fixed by:  https://github.com/celestiaorg/orchestrator-relayer/pull/101

**Review by evan-forbes — APPROVED** [review]

**Comment by SweeXordious** [comment]
@rahulghangas Can you please check this so that I can open up new PRs?

File: testing/evm_chain.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// EVMChain is a wrapped geth simulated backend which will be used to simulate an EVM chain.
+type EVMChain struct {
+	Auth         *bind.TransactOpts
+	GenesisAlloc core.GenesisAlloc
+	Backend      *backends.SimulatedBackend
+	Key          *ecdsa.PrivateKey
+	ChainID      uint64
+}
+
+func NewEVMChain(key *ecdsa.PrivateKey, chainID uint64) *EVMChain {
```
[the remark this answers is not in the document]
Comment (SweeXordious) [reply testing/evm_chain.go:24]: I have docs for that in local. After this is merged, new PRs containing that will come. Thanks for flagging this

File: testing/evm_chain.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			e.Backend.Commit()
+		}
+	}
+}
+
+func (e *EVMChain) Close() {
```
[the remark this answers is not in the document]
Comment (SweeXordious) [reply testing/evm_chain.go:69]: https://github.com/celestiaorg/orchestrator-relayer/pull/95#discussion_r1101178234

File: testing/testnode.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+	return &TestNode{
+		Context:         ctx,
+		DHTNetwork:      dhtNetwork,
+		CelestiaNetwork: celestiaNetwork,
+		EVMChain:        evmChain,
+	}
+}
+
+func (tn TestNode) Close() {
```
[the remark this answers is not in the document]
Comment (SweeXordious) [reply testing/testnode.go:47]: https://github.com/celestiaorg/orchestrator-relayer/pull/95#discussion_r1101178234

**Pull request merged**